make_prg/utils/seq_utils.py

```diff
@@ -234,3 +237,46 @@ def get_consensus_from_MSA(alignment: MSA) -> str:
             consensus_string_as_list.append(column.pop())
     consensus_string = "".join(consensus_string_as_list)
     return consensus_string
+
+
+def get_majority_consensus_from_MSA(alignment: MSA) -> str:
+    """
+    Produces a consensus string just based on the major base for each column.
+    Note: alignment needs to be upper-cased bases
+    """
+    all_seqs = "".join(get_alignment_seqs(alignment))
+    random_seed_for_this_alignment = hashlib.sha256(all_seqs.encode()).digest()
+    random.seed(random_seed_for_this_alignment)
+
+    # Initialize the consensus sequence as an empty string
+    consensus = ""
+
+    # Loop over the positions in the alignment
+    for i in range(alignment.get_alignment_length()):
+        # Count the residues at this position, ignoring gaps
+        pos_counts = Counter(
+            record.seq[i] for record in alignment if record.seq[i] != "-"
+        )
+
+        # If there are no residues other than gaps at this position, use a random base
+        if len(pos_counts) == 0:
+            consensus += random.choice("ACGT")
+            continue
+
+        # Find the residue with the highest count
+        max_residue, max_count = pos_counts.most_common(1)[0]
```

Comment:
What about the scenario where the two most common have the same count? Might be better to randomly select? The order in the `Counter` will be insertion order - which I guess you might be able to argue is randomised in a way...

Reply:
Yes, you are right, it is better to randomly select. Done in https://github.com/iqbal-lab-org/make_prg/pull/60/commits/2089324c0e2bfa68ae21a002e449b81642c560e9